exec/java-exec/src/main/java/org/apache/drill/exec/store/easy/json/JsonRecordWriter.java

```diff
@@ -82,6 +84,9 @@ public void init(Map<String, String> writerOptions) throws IOException {
     Path fileName = new Path(location, prefix + "_" + index + "." + extension);
     try {
       stream = fs.create(fileName);
+      // set storage strategy for folder and file
+      storageStrategy.apply(fs, fileName.getParent());
+      storageStrategy.apply(fs, fileName);
```

Comment:
Hmmm... The strategy has permission of "775" (permanent) and "700" (temporary). Those include executable. May be fine for the directory. Not so good for the data file. Maybe need an applyToFile and applyToDir? Or, let the storage strategy split the name?

Also, it is not clear if we are creating the directory here, or just changing an existing one. Seems we should honor existing perms on existing dirs; but set then only on new dirs. So, maybe have a mkdir() method that works like mkdirs(): create dir if needed (and imply permissions) or do nothing if the directory exists.

Does writer here create a directory? Will we know to remove it at end of session? We set delete on exit, but exit may be weeks away (for a long-lived Drillbit). Do we know to cascade deletes down into directories if the writer creates a partitioned file?